connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -255,8 +255,8 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig)
             allGroups.addAll(basicConfigDef.groups());
 
             ConnectorConfig connectorConfigToEnrich = (connector instanceof SourceConnector)
-                    ? new SourceConnectorConfig(plugins(), connectorConfig)
-                    : new SinkConnectorConfig(plugins(), connectorConfig);
+                    ? new SourceConnectorConfig(plugins(), connectorConfig, false)
```

Comment:
I see what you're saying. Sounds definitely better not to have to diverge between basic and connector-specific validation. I don't have too much background with the code around connector configs.